core/include/gnuradio-4.0/ClaimStrategy.hpp

```diff
@@ -24,12 +24,12 @@ struct NoCapacityException : public std::runtime_error {
 
 template<typename T>
 concept ClaimStrategy = requires(T /*const*/ t, const std::vector<std::shared_ptr<Sequence>> &dependents, const std::size_t requiredCapacity,
-        const std::make_signed_t<std::size_t> cursorValue, const std::make_signed_t<std::size_t> sequence, const std::make_signed_t<std::size_t> availableSequence, const std::size_t n_slots_to_claim) {
+        const std::make_signed_t<std::size_t> cursorValue, const std::make_signed_t<std::size_t> sequence, const std::make_signed_t<std::size_t> offset, const std::make_signed_t<std::size_t> availableSequence, const std::size_t n_slots_to_claim) {
     { t.hasAvailableCapacity(dependents, requiredCapacity, cursorValue) } -> std::same_as<bool>;
     { t.next(dependents, n_slots_to_claim) } -> std::same_as<std::make_signed_t<std::size_t>>;
     { t.tryNext(dependents, n_slots_to_claim) } -> std::same_as<std::make_signed_t<std::size_t>>;
     { t.getRemainingCapacity(dependents) } -> std::same_as<std::make_signed_t<std::size_t>>;
-    { t.publish(sequence) } -> std::same_as<void>;
+    { t.publish(offset, n_slots_to_claim) } -> std::same_as<void>;
```

Comment:
note: while this is an API change, it corresponds to the [API from disruptor-cpp](https://github.com/Abc-Arbitrage/Disruptor-cpp/blob/master/Disruptor/MultiProducerSequencer.h#L186) and with the current API i don't see a way to reliably publish more than one slot (apart from introducing additional state or other different API).

Reply:
OK.